flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -285,11 +285,7 @@ public StreamGraph generate() {
 			alreadyTransformed.put(transform, transformedIds);
 		}
 
-		if (transform.getBufferTimeout() >= 0) {
-			streamGraph.setBufferTimeout(transform.getId(), transform.getBufferTimeout());
-		} else {
-			streamGraph.setBufferTimeout(transform.getId(), defaultBufferTimeout);
```

Comment:
Can you explain the reasoning behind removing this branch?

I think we still need it, because if the user doesn't set a timeout for transformation then default job timeout should be used.

Reply:
We should only set the default timeout for streaming job by design. But in this procedure we can not determine the `ResultPartitionType` for properly setting the default value, so I removed that path here.

But the default timeout still validates in the last procedure while generating `StreamConfig` by `#setDefaultTimeout()`.